.github/workflows/lint_python.yml

```diff
@@ -1,17 +1,25 @@
 name: lint_python
-on: [pull_request, push]
+on:
+  pull_request:
+  push:
+    branches: [master]
 jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: pip install pre-commit
+      - run: pre-commit --version
+      - run: pre-commit install
+      - run: pre-commit autoupdate
+      - run: pre-commit run --all-files
   lint_python:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
-      - run: pip install bandit black codespell flake8 isort mypy pytest pyupgrade safety
-      - run: bandit --recursive --skip B101,B108,B301,B403,B404,B603 .
-      - run: black --check .
-      - run: codespell --ignore-words-list=nd,reacher,thist,ths -w
-      - run: flake8 --ignore=E203,E402,E712,E722,E731,E741,F401,F403,F405,F524,F841,W503
-                    --max-complexity=30 --max-line-length=456 --show-source --statistics
+      - run: pip install isort mypy pytest pyupgrade safety
       - run: isort --check-only --profile black . || true
```

Comment:
Do you wanna move `isort` into `pre-commit` as well (it works super nicely for sorting your imports when you run `pre-commit` locally)?

Reply:
pytest fails after running isort.